docs/pages/project/contributing/contributing-cli.md

```diff
@@ -128,6 +128,25 @@ When designing for the command line interface, ask and consider the following qu
 
 You will need to provide a short and long description of the command for the help pages and also for the Meshery Documentation.
 
+## Refactoring Mesheryctl Code
+
+The mesheryctl code needs to be refactored for the ease of management, to improve the readbility of the code, for making it easier for using the utility packages. 
+
+For now, the meshconfig has been refactored to have getters and setters which can further be used to manipulate any config. For example, reading or changing the context version has now been a lot easier with the help of specific getters and setters, a central function has also been added to write the config.yaml file.
+
+The workflow for changing the contexts would be:
+- Change the context struct
+- Update the config struct with the new context
+- Write the new config to the config.yaml file
+
+Here are some examples of how the getters and setters of some defined structs are being used:
+
+Let us take an example of the `struct Context`, having the parameter `Token`, the code has been refactored to use `func(ctx *Context) GetToken()` for returning the token of the current context, and `func(ctx *Context) SetToken()` has been used for setting the token of the current context.
+
+Similarly, for the `struct Mesheryctlconfig`, having the `Tokens` parameter has a function named `func (mc *MesheryCtlConfig) GetTokens() *[]Token` which is being used to return the tokens present in the config file.
+
+There are more similar changes made, the changes can be viewed in `/mesheryctl/internal/cli/root/config/config.go` file and all the usages of the config/struct throughout the codebase has been modified to use the new struct functions.
```

Comment:
There's no such thing as "more similar".

Reply:
```suggestion
There are changes made in the codebase, the changes can be viewed in `/mesheryctl/internal/cli/root/config/config.go` file and all the usages of the config/struct throughout the codebase has been modified to use the new struct functions.
```

Reply:
Thank you, @alphaX86. This is an improvement. Now, the paragraph needs rewritten to not walk the reader through the history of what has happened in mesheryctl - - it doesn't matter. They reader needs to understand the current organizations of structs, their purpose, how to use them, their behavior, and see an example.

Reply:
Agreed 👍